share/rocm/cmake/header_template.h.in

```diff
@@ -8,7 +8,7 @@
 #ifndef ROCM_@ITEM_GUARD@_GAVE_WARNING
 #define ROCM_@ITEM_GUARD@_GAVE_WARNING
 
-#warning This file is deprecated. Use the header file from @header_location@ by using #include <@correct_include@>
+#pragma message("This file is deprecated. Use the header file from @header_location@ by using #include <@correct_include@>")
```

Comment:
@pfultz2 we think its better to give #pragma message as the first step. If any of the application using our header file have enabled **-Werror** option in the compiler, #warning will result in compilation error. 
Better give this deprecated message for few releases and then can change it to #error or #pragma warning(error).

Reply:
> If any of the application using our header file have enabled -Werror option in the compiler

Thats what happens when using `[[deprecated(message)]]` as well, so it should be a warning to be consistent with other deprecation warnings.

Reply:
The `#warning` should probably also be behind an `#elif defined(__GNUC__)` check.

Reply:
Using `#pragma message ....` is clearly the correct thing to do as the first step. **We** are changing things so that previously perfect code needs to be updated. **We** should not cause existing builds to fail for no cause.

If you want to do it the hard way then this is about as good as you can get (just preprocessing a file still gets a non-zero exit code).

     #pragma GCC diagnostic push
     #pragma GCC diagnostic warning "-Wcpp"
     #warning .....
     #pragma GCC diagnostic pop

Personally I think this code is over engineered, addressing a short term problem. If this is a real long term concern then there ought to be a version number being passed in and compared against, so that the choice can be made if this should be forced to informative only, or else being issued at a `warning` level so the **user** of the code can decide if they want to treat it as an `error` or not.

Reply:
Thanks @amd-isparry for the suggestion. With -Wcpp option its working with gcc, amd clang and clang++. But its not working for g++. Seems its a known issue with g++ . https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53431
Is there any better flag to use here? Or g++ can be ignored in our case?

Reply:
I don't think we can ignore g++, so better to ignore my suggestion.